### ISSUE TITLE
Always returns the same films

### ISSUE THREAD

**Issue by ajk1982** [jjoej15/letterboxd-recs#1]
Hi,

I think this is excellent, thanks for sharing. One issue I have is the film recommendations are always the same. You can put in any two names for a blended approach and you always get the same films maybe one or two differences in the first ten results:
<img width="1027" alt="Screenshot 2024-10-20 at 22 27 22" src="https://github.com/user-attachments/assets/293647b8-d042-4d47-89a9-003259be5ea9">

**Comment by jjoej15** [comment]
Yes, I certainly noticed this issue when developing this months ago. It's likely a result of the underlying model being trained on a relatively small sample size of about 5000 users, and it's completely blind to most attributes of films like date released, genre, directors, etc. It only makes predictions based on how users similar to you rate films, and since films like Parasite, Before Sunrise, etc. all get very high ratings from most everybody, it almost always recommends them to you too. This is why I implemented the popularity and genre filters, to try and counteract this to a degree. Might have to try and add more parameters to train the model with, or maybe implement some way that a smaller version of the model is built dynamically when running the program so things feel more random. Thanks for checking out my project and pointing this out!

**Comment by ajk1982** [comment]
Ah yes it is an interesting problem. I love the blend mode. I have been playing around with the python locally and added the ability to blend more than two users. I also added a release year filter!
